### PR TITLE
fix: update horizontal-overflow default height to null instead of 0px

### PR DESCRIPTION
**Pull request by JamesBurnside**
# Description

Set Horizontal-Overflow default height to unset instead of 0px.

## Motivation & context

#2217
Having 0px can unintentionally hide children.
This is more apparent when server side rendering. The issue is that this.state.itemsHeight is never set to this.getItemMaxHeight();
This is because componentDidMount and itemsLoaded are never triggered when server side rendering and hence itemHeight stays at a default of 0px.

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [x] **Bug fix**: A change that fixes an issue, link to the issue above.
- [ ] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [x] This change causes current functionality to break.

Components depending on this being set to 0px may experience different functionality.

## Process & policy checklist

- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/en/contributing/standards) for this project.

### CODE REVIEW

File: packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.tsx [hunk 1]
```diff
@@ -118,7 +118,10 @@ class HorizontalOverflow extends Foundation<
             >
                 <div
                     style={{
-                        height: `${this.state.itemsHeight}px`,
+                        height:
+                            this.state.itemsHeight !== null
+                                ? `${this.state.itemsHeight}px`
+                                : "unset",
```
Comment (chrisdholt) [inline packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.tsx]: I don't believe `unset` is supported whatsoever in IE. https://developer.mozilla.org/en-US/docs/Web/CSS/unset

File: packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.tsx [hunk 2]
```diff
@@ -28,7 +28,7 @@ export enum ButtonDirection {
 }
 
 export interface HorizontalOverflowState {
-    itemsHeight: number;
+    itemsHeight?: number;
```
Comment (nicholasrice) [inline packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.tsx]: What is the reason for making `itemsHeight` an optional property? It always exists and is set. Is it more accurate to change to `itemsHeight: number | null` to make clear that `null` is a valid value of this property?

File: packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.tsx (hunk 1 above)
Comment (nicholasrice) [reply packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.tsx]: We should validate it works correctly in IE, but I expect it does - the property should be ignored if the value is unrecognized, which *should* have the same effect as "unset". Another option would be to assign `height` to `undefined` and let React filter the property out.

**Comment by JamesBurnside** [comment]
> You mention that having 0px can unintentionally hide children - this concern is specific to rendering server side, correct?

I assume it is visibly an issue on client because itemsHeight is set to a new value on ComponentDidMount (which isn't fired when server side rendering) so it wouldn't really be seen on client. Let me check

File: packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.tsx (hunk 1 above)
Comment (JamesBurnside) [reply packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.tsx]: If React filters out undefined then that is likely the better alternative (that have the possibility of IE give a warning/undefined behavior on a property)

File: packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.tsx (hunk 2 above)
Comment (JamesBurnside) [reply packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.tsx]: The issue currently is that itemsHeight defaults to 0px as it is expected to be updated by componentDidMount() or by itemsLoaded() callbacks.
However these are not triggered server side so itemsHeight has an incorrect value of 0px (and technically incorrect at points on client side but a non-issue as componentDidMount/itemsLoaded will update the state)
So I was making it optional to indicate that it hadn't been set. An alternative would be to use NaN as the default instead

Comment (nicholasrice) [reply packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.tsx]: It *is* always set though right? Sorry if I'm missing something, but it appears it is always set to a pixel value or `null`, meaning there is never a case where the property *doesn't exist*

Comment (JamesBurnside) [reply packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.tsx]: Ah new to typescript, my bad thought the `?` meant nullable as opposed to optional. Will remove the `?`.

**Comment by JamesBurnside** [comment]
@janechu for review

File: packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.tsx [hunk 3]
```diff
@@ -118,7 +118,10 @@ class HorizontalOverflow extends Foundation<
             >
                 <div
                     style={{
-                        height: `${this.state.itemsHeight}px`,
+                        height:
+                            this.state.itemsHeight !== null
+                                ? `${this.state.itemsHeight}px`
+                                : "undefined",
```
Comment (janechu) [inline packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.tsx]: There is no such state as "undefined" for height, I think it should default to "auto" per https://developer.mozilla.org/en-US/docs/Web/CSS/height

Comment (JamesBurnside) [reply packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.tsx]: Sounds good, I'll update to auto now 👍

File: packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.tsx (hunk 1 above)
Comment (JamesBurnside) [reply packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.tsx]: Updated to auto per @janechu's comment, resolving this conversation to only have one thread open on the issue

**Comment by chrisdholt** [comment]
Thanks @JamesBurnside for the PR!

**Pull request merged**